core/src/main/java/org/testcontainers/containers/traits/TraitsSupport.java

```diff
@@ -0,0 +1,71 @@
+package org.testcontainers.containers.traits;
+
+import lombok.NonNull;
+import org.testcontainers.containers.TestContainer;
+import org.testcontainers.utility.SelfReference;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public interface TraitsSupport<SELF extends TestContainer<SELF>> extends SelfReference<SELF> {
+
+    Comparator<Trait> TRAIT_COMPARATOR = (obj1, obj2) -> {
+        if (obj1 instanceof Comparable && obj2 instanceof Comparable)
+        {
+            Comparable cmp1 = (Comparable)obj1;
+            return cmp1.compareTo(obj2);
+        }
+
```

Comment:
traits might not implement Comparable, so we shouldn't fail, but handle such case